docs/key-issues/multiple_architectures.md

```diff
@@ -0,0 +1,129 @@
+application, a version must be provided for each supported CPU architecture. A
+directory layout convention exists for providing a binary for each platform,
+with the same library name. This yields an independent final binary (APK) for
+each CPU architecture. When running locally, a CPU-specific APK will be
+uploaded to the simulator or test device.
+
+To simplify the process of distributing the application, at time of publication,
+a single Android App Bundle (AAB) is generated from the multiple CPU-specific
+APKs. This AAB contains binaries for all platforms that can be uploaded to an
+app store.
+
+When an end-user requests the installation of an app, the app store strips out the
+binary that is appropriate for the end-user's device.
+
+### Fat binaries
+
+Apple has taken the approach of "fat" binaries. A fat binary is a single
+executable or library artefact that contains code for multiple CPU
+architectures.
+
+Fat binaries can be compiled in two ways:
+
+1. **Single pass** Apple has modified their compiler tooling with flags that
+   allow the user to specify a single compilation command, and instruct the
+   compiler to generate multiple output architectures in the output binary
+2. **Multiple pass** After compiling a binary for each platform, Apple provides
+   a call named `lipo` to combine multiple single-architecture binaries into a
+   single fat binary that contains all platforms.
+
+At runtime, the operating system loads the binary slice for the current CPU
+architecture, and the linker loads the appropriate slice from the fat binary of
+any dynamic libraries.
+
+On macOS ARM hardware, Apple also provides Rosetta as a support mechanism; if a
+user tries to run an binary that doesn't contain an ARM64 slice, but *does*
+contain an x86-64 slice, the x86-64 slice will be converted at runtime into an
+ARM64 binary. Complications can occur when only *some* of the binary is being
+converted (e.g., if the binary being executed is fat, but a dynamic library
+isn't).
+
+iOS has an additional complication of requiring support for mutiple *ABIs* in
+addition to multiple CPU archiectures. The ABI for the iOS simulator and
+physical iOS devices are different; however, ARM64 is a supported CPU
+architecture for both. As a result, it is not possible to produce a single fat
+library that supports both the iOS simulator and iOS devices. Apple provides an
+additional structure - the `XCFramework` - as a wrapper format for packaging
+libraries that need to span multiple ABIs. When developing an application for
+iOS, a developer will need to install binaries for both the simulator and
+physical devices.
+
+## Potential solutions or mitigations
+
+Python currently provides `universal2` wheels to support x86_64 and ARM64 in a
+single wheel. This is effectively a "fat wheel" format; the `.dylib` files
+contained in the wheel are fat binaries containing both x86_64 and ARM64 slices.
```

Comment:
I would say that `universal2` should go in the "current state" rather than "potential solutions" section. Also, I'd like to point out that `universal2` is not a good idea for Python packages - its introduction seemed to be motivated mainly by "this is easiest and we're in a hurry", and the complications and disk space / bandwidth costs it entails are in general not justified (certainly not for wheels on PyPI).